R/get_lifetable.R

```diff
@@ -0,0 +1,140 @@
+                           years.end=NA, 
+                           ages.start=0,
+                           ages.end=NA){
+
+  stopifnot("Country must be a character." = is.character(country))
+  stopifnot("Gender must be a character."=is.character(gender))
+  stopifnot("Years.start must be numeric"=is.numeric(years.start))
+  stopifnot("Ages.start must be numeric"=is.numeric(ages.start))
+
+  url<-"https://population.un.org/wpp/Download/Files/1_Indicators%20(Standard)/CSV_FILES/WPP2019_Life_Table_Medium.csv"
+
+  DT = vroom::vroom(url)
+
+  # check format of country input and reformat to country name
+  
+  if(country != "all") {
+    # convert iso3c codes
+    if(country %in% countrycode::codelist$iso3c){
+      country <- countrycode::countrycode(country, origin="iso3c", destination="country.name")
+    } 
+    # filter by country - has tolower in case someone misspelled something
+    if(tolower(country) %in% tolower(DT$Location)){
+      DT = DT[tolower(DT$Location) == tolower(country),]
+    }
+  }
+  # another thing: there are some regions that are included that aren't countries per se - should we filter these out or keep them?
+  
+  # add gender check
+  gender <- tolower(gender)
+  if(gender!="all"){ # if user says all, will return female, male and total
+    if(gender!="both"){
+      gender <- ifelse(startsWith(gender, "f"), "Female", "Male")
+    } else {
+      gender<-"Total"
+    }
+    DT=DT[DT$Sex==gender,]
+  }
+  # check years start & end
+  # chop up time into time into time.start and time.end
+  DT <- as.data.frame(DT)
+  DT <- tidyr::separate(DT, Time, c("TimeStart", "TimeEnd"))
+  DT$TimeStart<-as.numeric(DT$TimeStart)
+  DT$TimeEnd<-as.numeric(DT$TimeEnd)
+  
+  if(is.numeric(years.start)){ 
+    DT=DT[DT$TimeStart>=years.start,]
+    DT=DT[DT$TimeEnd>years.start,]
+  } 
+  if(!is.na(years.end)){
+    DT=DT[DT$TimeStart<years.end,]
+    DT=DT[DT$TimeEnd<=years.end,]
+  } 
+  
+  # check ages start & end
+  if(is.numeric(ages.start)){
+    if(ages.start %in% DT$AgeGrpStart){
+      DT<-DT[DT$AgeGrpStart>=ages.start,]
+    } else { # age start is not at the natural bound of the age group, so return the age group including this age
+      while(!(ages.start %in% DT$AgeGrpStart)){
+        ages.start<-ages.start-1
+      }
+      DT<-DT[DT$AgeGrpStart>=ages.start,]      
+    }
+  }
+  
+  if(!is.na(ages.end)){
+    DT<-DT[DT$AgeGrpStart<ages.end,]
+  }
+
+  # return life table without certain columns as dataframe
+  DT<-DT[, c(2,5:7,9:22)]
+  return(DT) # return as dataframe
+}
+
+#' Get deathrates if input a life table returned from the [get_lifetable()] function
+#' @param data data frame of life table data returned from [get_lifetable()]
+#'
+#' @keywords UN, demography, download, population, life table, deathrates, mx, ax, dx, gender, age groups
+#' @return Data frame of death rates
+#' @export
+#' 
+#' @examples
+#' extract_deathrates(DT)
+extract_deathrates <- function(data){
```

Comment:
I think this would be the function to convert the data to `malariasimulation` intput - is that right?
If so we should make sure we are consistent with naming conventions and perhaps link in the documentation to the relevant function and/or vignette in malariasimualation?